agent/src/datafile.rs

```diff
@@ -490,6 +565,40 @@ impl DataFile {
         Ok(())
     }
 
+    /**
+     * Mark a particular running snapshot as destroyed.
+     */
+    pub fn destroyed_rs(
+        &self,
+        region_id: &RegionId,
+        snapshot_name: &str,
+    ) -> Result<()> {
+        let mut inner = self.inner.lock().unwrap();
+
+        let mut rs = inner
+            .running_snapshots
+            .get_mut(region_id)
+            .unwrap()
+            .get_mut(snapshot_name)
+            .unwrap();
+
+        let nstate = State::Destroyed;
+
+        info!(
+            self.log,
+            "running snapshot {} state: {:?} -> {:?}",
+            rs.id.0,
+            rs.state,
+            nstate,
+        );
+        rs.state = nstate;
+
+        // XXX shouldn't this remove the record?
```

Comment:
If we don't remove the record, then any attempt by Nexus to provision a running snapshot with the same name for the same region wouldn't work, it would return the existing record with `state == destroyed`. We currently keep the record around for debugging purposes.